quickcheck-dynamic/src/Test/QuickCheck/StateModel.hs

```diff
@@ -409,6 +379,57 @@ instance forall state. StateModel state => Arbitrary (Actions state) where
               | otherwise = ps : map (p :) (go ps)
          in go acts
 
+newtype MoreActions state = MoreActions {getMoreActions :: Actions state}
+
+instance Show (Actions state) => Show (MoreActions state) where
+  show = show . getMoreActions
+
+instance StateModel state => Arbitrary (MoreActions state) where
+  arbitrary = MoreActions <$> generateActionsWithOptions (defaultGenActionsOptions{genOptLengthMult = 10})
```

Comment:
What if a user wants to generate really long sequences, eg. with a multiplicative factor of 50 or 100 or 1000? Would you advise to use `MoreActions (MoreActions ...)`?

Reply:
That wouldn't work, the types are wrong. For that use-case I think one ought to use `generateActionsWithOptions`. But this is the ugly part of the design that I'm not sure about. Maybe having an auxilliary class for `arbitraryWithOptions` would allow us to set up the types in such a way that every wrapper changes the options locally.

Reply:
Having a fixed factor 10 does seem a bit inflexible. Maybe a type-level `KnownNat` for the factor?

Reply:
yuck?